### PR TITLE
documenting some undocumented classes and methods

### PR DESCRIPTION
**Pull request by tiagmoraismorgado**
documenting some undocumented classes and methods (or otherwise adding examples, in case they don't exist)

### CODE REVIEW

File: HelpSource/Classes/Array.schelp [hunk 1]
```diff
@@ -57,37 +57,47 @@ copymethod:: ArrayedCollection *series
...
 copymethod:: ArrayedCollection -clipAt
 
 copymethod:: ArrayedCollection -wrapAt
 
 copymethod:: ArrayedCollection -foldAt
 
-copymethod:: ArrayedCollection -clipPut
+copymethod:: ArrayedCollection -slice
+
+method:: put
```
Comment (jamshark70) [inline HelpSource/Classes/Array.schelp]: Should be `method:: clipPut` to match the example...? Also this doesn't explain how `clipPut` is different from `put`.

File: HelpSource/Classes/DrawGridX.schelp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+CLASS:: DrawGridX
+summary:: A sub-method of DrawGrid, used for setting up its horizontal size
```
Comment (jamshark70) [inline HelpSource/Classes/DrawGridX.schelp]: Subclass... there's no such thing as a "sub-method."

File: HelpSource/Classes/DrawGridX.schelp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+categories:: GUI>Accessories
+related:: Reference/plot, Classes/GridLines, Classes/Plotter, Classes/UserView, Classes/DrawGrid
```
Comment (jamshark70) [inline HelpSource/Classes/DrawGridX.schelp]: Does `Reference/plot` actually work when you click on it? The lowercase looks suspicious to me.

Test links before pushing, please.

File: HelpSource/Classes/DrawGridX.schelp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+DESCRIPTION::
+DrawGridX, is a method instantiated by DrawGrid Class, in its init functon, to set the horizontal size of a GridUI Element, just as seen in the code below
```
Comment (jamshark70) [inline HelpSource/Classes/DrawGridX.schelp]: *Not a method!!!*

This description is quite unclear.

File: HelpSource/Classes/DrawGridX.schelp [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+CLASSMETHODS::
+
+METHOD:: copy
+used for copying this instance
+
+EXAMPLES::
+
+code::
+// taken from Grid.sc class file
+init { arg bounds,h,v;
```
Comment (jamshark70) [inline HelpSource/Classes/DrawGridX.schelp]: Examples should be code that the user can execute directly. The only time an example should be class-library code is if the help file is about writing classes. I disagree with this example.

File: HelpSource/Classes/DrawGridY.schelp [hunk 6]
```diff
@@ -0,0 +1,28 @@
+CLASS:: DrawGridY
+summary:: A sub-method of DrawGrid, used for setting up its vertical size
```
Comment (jamshark70) [inline HelpSource/Classes/DrawGridY.schelp]: Same comments as DrawGridX, throughout.

File: HelpSource/Classes/QuarksGui.schelp [hunk 7]
```diff
@@ -0,0 +1,23 @@
+class:: QuarksGui
+summary:: Allows Quarks Installation
```
Comment (jamshark70) [inline HelpSource/Classes/QuarksGui.schelp]: Better: "A graphical interface to manage Quarks extension packages"

File: HelpSource/Classes/QuarksGui.schelp [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+related:: Classes/Quarks
+
+Description::
+
+Quarks are packages of SuperCollider code containing classes, extension methods, documentation and server UGen plugins. The QuarksGui class acts as a frontend to Quarks, being instanted in the Quarks Class, to call its UI/Frontend, just as you can see if you open the its respective *.sc  file. The created UI/Fronted window allows to manage downloading these packages and installing or uninstalling them.
```
Comment (jamshark70) [inline HelpSource/Classes/QuarksGui.schelp]: Users generally don't use this class directly. IMO the only documentation required is to state that users create a QuarksGui by running `Quarks.gui`, and link to the main Quarks documentation. Users are not expected to call any methods on this class directly, so we should not waste effort documenting what they aren't supposed to use anyway.

File: HelpSource/Classes/VolumeGui.schelp [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+v.min;
+v.max;
+v.volume = rrand(-50, 5);
+v.setVolumeRange(-90, 8);
+v.mute;
+v.unmute;
+
+VolumeGui.new(v, 16, 320)
+
+// code needs fixing (just tried to make sense of the class intuitively wise
```
Comment (jamshark70) [inline HelpSource/Classes/VolumeGui.schelp]: Maybe ask about the example on the mailing list before pushing?

File: HelpSource/Classes/Array.schelp [hunk 10]
```diff
@@ -57,37 +57,47 @@ copymethod:: ArrayedCollection *series
...
 InstanceMethods::
 
 copymethod:: ArrayedCollection -at
 
-copymethod:: ArrayedCollection -put
+method:: put
+Put item at index, replacing what is there.
+code::
+x=[1, 2,3 , 4, 5]
+x.put(rrand(0, 4), [33, 77, 99, 322, 555, 666, 999].choose)
```
Comment (vivid-synth) [inline HelpSource/Classes/Array.schelp]: The randomness here seems unnecessary -- I think this would be clearer without it

File: HelpSource/Classes/DrawGridX.schelp (hunk 3 above)
Comment (jamshark70) [inline HelpSource/Classes/DrawGridX.schelp]: I just checked, Reference/plot should be fine -- there is a plot.schelp file. Sorry for noise.

File: HelpSource/Classes/Array.schelp [hunk 11]
```diff
@@ -57,37 +57,47 @@ copymethod:: ArrayedCollection *series
...
-copymethod:: SequenceableCollection *fib
+copymethod:: Collection *rand
 
+copymethod:: Collection *rand2
 
+copymethod:: Collection *linrand
 
+copymethod:: Collection *exprand
 
+copymethod:: Collection *fib
```
Comment (tiagmoraismorgado) [inline HelpSource/Classes/Array.schelp]: that makes sense. probably asking help to someone to do that would be a great idea, as I can't quite figure out how to do that on my own

File: HelpSource/Classes/VolumeGui.schelp (hunk 9 above)
Comment (tiagmoraismorgado) [inline HelpSource/Classes/VolumeGui.schelp]: you are right

Comment (tiagmoraismorgado) [inline HelpSource/Classes/VolumeGui.schelp]: if someone can provide help fixing the code, I would be quite thankful

**Comment by tiagmoraismorgado** [comment]
I've gone throughout most of the **comments** of @jamshark70. **_two things missing:_**

**a)** - help file to **plot supercollider method**
**b)** - quarks **aggregating the array copy method** business

regarding a): I can do it myself and either add it to this pull request (need someone to explain me how to do so), or add a separate pull request

**regarding b)**: definitely need some help to do so: @jamshark70  @vivid-synth  @snappizz would you be into giving me a hand with this?

**looking forward, kind regards**
**xx
T.**

File: HelpSource/Classes/Array.schelp (hunk 11 above)
Comment (jamshark70) [inline HelpSource/Classes/Array.schelp]: I think you have to do it by hand here... git can revert a whole commit. It can't revert changes to one file that is part of a commit. Your first commit includes changes to a lot of files, and there's no way to separate them after the fact.

Generally one commit should be for one set of closely related changes. Then, you can put several commits together into one branch, for a PR.

You have a bunch of unrelated changes in one commit. Probably not the best way.

It might be better to fix the problems, then delete this branch and start over with the PR for a cleaner history.

**Comment by brianlheim** [comment]
Sorry, I'm not interested in helping with this PR, but I will look it over for typos when it's ready to merge.

File: HelpSource/Classes/Array.schelp [hunk 12]
```diff
@@ -69,25 +69,35 @@ copymethod:: SequenceableCollection *exprand
...
 InstanceMethods::
 
 copymethod:: ArrayedCollection -at
 
-copymethod:: ArrayedCollection -put
+method:: put
+Put item at index, replacing what is there.
+code::
+x=[1, 2,3 , 4, 5]
+x.put(rrand(0, 4), 8)
```
Comment (vivid-synth) [inline HelpSource/Classes/Array.schelp]: I'd remove the ```rrand``` as it makes the example more complicated. I'd also put a note about what the result is, like:

    x // result: [ 1, 8, 3, 4, 5 ]

File: HelpSource/Classes/Array.schelp [hunk 13]
```diff
@@ -69,25 +69,35 @@ copymethod:: SequenceableCollection *exprand
...
 copymethod:: ArrayedCollection -clipAt
 
 copymethod:: ArrayedCollection -wrapAt
 
 copymethod:: ArrayedCollection -foldAt
 
-copymethod:: ArrayedCollection -clipPut
+copymethod:: ArrayedCollection -slice
+
+method:: clipPut
```
Comment (vivid-synth) [inline HelpSource/Classes/Array.schelp]: Why the change from "copymethod:: ArrayedCollection -clipPut"?

**Comment by brianlheim** [comment]
Ping. @tiagmoraismorgado are you still planning to work on this?

**Comment by brianlheim** [comment]
Also, I noticed these commits are being merged from your master branch. In the future, remember that it's better form to make changes on a different branch, and then request to pull from that branch. Thanks!

**Comment by tiagmoraismorgado** [comment]
greetings everyone. i tried to make the suggested improvements, as suggested. shall we actually merge this? please help me to finish the things that are left to merge, and proceed with merging this PR

File: HelpSource/Classes/Array.schelp [hunk 14]
```diff
@@ -69,18 +69,21 @@ copymethod:: SequenceableCollection *exprand
...
-
 InstanceMethods::
 
 copymethod:: ArrayedCollection -at
 
-copymethod:: ArrayedCollection -put
+method:: put
+Put item at index, replacing what is there.
+code::
+x=[1, 2,3 , 4, 5]
```
Comment (gusano) [inline HelpSource/Classes/Array.schelp]: @tiagmoraismorgado Spacing is wrong here

File: HelpSource/Classes/Array.schelp [hunk 15]
```diff
@@ -89,6 +92,13 @@ copymethod:: ArrayedCollection -foldAt
 
 copymethod:: ArrayedCollection -clipPut
 
+method:: clipPut
+Put item at index, replacing what is there.
+code::
+x=[1, 2,3 , 4, 5]
```
Comment (gusano) [inline HelpSource/Classes/Array.schelp]: @tiagmoraismorgado Same spacing error

File: HelpSource/Classes/DrawGridX.schelp [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+		var w;
+		x = DrawGridX(h);
+		y = DrawGridY(v);
+		this.bounds = bounds;
+		this.font = Font( Font.defaultSansFace, 9 );
+		this.fontColor = Color.grey(0.3);
+		this.gridColors = [Color.grey(0.7),Color.grey(0.7)];
+	}
```
Comment (gusano) [inline HelpSource/Classes/DrawGridX.schelp]: @tiagmoraismorgado IMHO This class code example does not really fit in an help file

File: HelpSource/Classes/DrawGridY.schelp [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+// taken from Grid.sc class file
+init { arg bounds,h,v;
+		var w;
+		x = DrawGridX(h);
+		y = DrawGridY(v);
+		this.bounds = bounds;
+		this.font = Font( Font.defaultSansFace, 9 );
+		this.fontColor = Color.grey(0.3);
+		this.gridColors = [Color.grey(0.7),Color.grey(0.7)];
+	}
```
Comment (gusano) [inline HelpSource/Classes/DrawGridY.schelp]: @tiagmoraismorgado Same remark about the class code example

File: HelpSource/Classes/VolumeGui.schelp [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+DESCRIPTION::
+VolumeGui is an abstraction method used by Volume to create a graphic user interface. If one looks into the volume source file, and look into the gui method, one may find the VolumeGui abstraction being used to instantiate the Volume UI Frontend.
+
+CLASSMETHODS::
+method:: new
+Creates a new window for VolumeGui new window when instanced
+
+CODE::
+v = s.volume;
```
Comment (gusano) [inline HelpSource/Classes/VolumeGui.schelp]: @tiagmoraismorgado Again, about the code example.. Maybe don't add examples for abstract classes? Or as @jamshark70 said, ask on the list first before submitting the PR?

File: SCClassLibrary/Common/Collections/SequenceableCollection.sc [hunk 19]
```diff
@@ -64,6 +64,16 @@ SequenceableCollection : Collection {
 		});
 		^obj
 	}
+	*recursiveFormula { arg size, a=0.0;
```
Comment (nuss) [inline SCClassLibrary/Common/Collections/SequenceableCollection.sc]: What's this method supposed to do? It seems to be not only slightly unrelated to the thread's topic "documenting some undocumented classes and methods"...

File: SCClassLibrary/Common/Collections/SequenceableCollection.sc [hunk 20]
```diff
@@ -64,6 +64,16 @@ SequenceableCollection : Collection {
...
+		var i=0, divisor = 3, divisorDec = 0.9;
+		var obj = this.new(size);
+		while { i < size }{
+			obj.multiply((a/divisor));
+			divisorDec -= 0.1;
```
Comment (nuss) [inline SCClassLibrary/Common/Collections/SequenceableCollection.sc]: Operators like `+=`, `-=`, `*=`, `/=` don't exist in sclang. The only correct way to decrement a number by 0.1 would be
`number = number - 0.1;`
Have you ever tested your method before making a pull request?

**Comment by brianlheim** [comment]
I am closing this because (1) @tiagmoraismorgado seems to have abandoned work on this, (2) there are too many problems here for the PR to be easily salvaged, and (3) it's already eaten up too much developer time.
